### PR TITLE
adding a check to see if hotelOwner is changing its hotel

### LINKED ISSUE #96: hotel owner should update/delete etc only hotels that belong to them (effort 1)

### PR DESCRIPTION
**Pull request by ibednorz**

### CODE REVIEW

File: services/ownerService.js [hunk 1]
```diff
@@ -41,13 +42,18 @@ exports.addHotel = async (data) => {
   return hotel
 }
 
-exports.updateHotel = async (id, data) => {
-  const hotelUpdate = await Hotel.findByIdAndUpdate(id, data)
+exports.updateHotel = async (id, data, req) => {
```
Comment (pdybowski) [inline services/ownerService.js]: Do these params matches arguments that are passed when we call this function? I mean here we have req and where we call this function we pass req.user._id

File: services/ownerService.js [hunk 2]
```diff
@@ -71,7 +77,11 @@ exports.deleteHotel = async (ownerId, id, isForceDelete) => {
   await Hotel.findByIdAndDelete(id)
 }
 
-exports.deleteReservation = async (id) => {
+exports.deleteReservation = async (id, req) => {
+  if (!isObjIdEqualToMongoId(req.user._id, reservation.hotel)) {
```
Comment (pdybowski) [inline services/ownerService.js]: What is reservation? We didn't declare it lol

**Review by pdybowski — COMMENTED** [review]
Did you test it?

**Comment by pdybowski** [comment]
still waiting for fixes

File: services/ownerService.js [hunk 3]
```diff
@@ -43,13 +43,18 @@ exports.addHotel = async (data) => {
   return hotel
 }
 
-exports.updateHotel = async (id, data) => {
-  const hotelUpdate = await Hotel.findByIdAndUpdate(id, data)
+exports.updateHotel = async (id, data, userId) => {
+  let hotel = await Hotel.findById(id)
+  if (!isObjIdEqualToMongoId(userId, hotel.ownerId)) {
```
Comment (pdybowski) [inline services/ownerService.js]: First we must check if hotel exists. Then if it belongs to the owner..

File: services/ownerService.js [hunk 4]
```diff
@@ -43,13 +43,18 @@ exports.addHotel = async (data) => {
   return hotel
 }
 
-exports.updateHotel = async (id, data) => {
-  const hotelUpdate = await Hotel.findByIdAndUpdate(id, data)
+exports.updateHotel = async (id, data, userId) => {
+  let hotel = await Hotel.findById(id)
```
Comment (pdybowski) [inline services/ownerService.js]: Check something like findOneAndUpdate. You should be able to check if hotel exists and belongs to the hotel owner.

**Review by pdybowski — CHANGES_REQUESTED** [review]
Nie No zle. Masz zmienna która nie jest nigdzie zdefiniowana. Sprawdzasz czy hotel istnieje później niż sprawdzasz czy należy do hotel ownera. Ten kod nie zadziała jak powinien...

File: controllers/ownerController.js [hunk 5]
```diff
@@ -47,7 +47,19 @@ exports.deleteHotel = async (req, res, next) => {
   try {
     const { forceDelete } = req.query
     const isForceDelete = forceDelete === 'true'
-    await deleteHotel(req.user, req.params.id, isForceDelete)
+    await deleteHotel(req.user._id, req.params.id, isForceDelete)
+    res.sendStatus(200)
+  } catch (error) {
+    if (error instanceof mongoose.Error.CastError) {
```
Comment (bartoszbialecki) [inline controllers/ownerController.js]: here should be only next(error) like I changed earlier

File: services/ownerService.js [hunk 6]
```diff
@@ -43,15 +43,21 @@ exports.addHotel = async (data) => {
...
+  let hotelUpdate = await Hotel.findById(id)
 
   if (!hotelUpdate) {
     throw new BadRequestError('Hotel not found.')
   }
-  const hotel = await Hotel.findById(id)
 
-  return hotel
+  if (!isObjIdEqualToMongoId(userId, hotel.ownerId)) {
+    throw new ApiError(403, 'Forbidden')
```
Comment (bartoszbialecki) [inline services/ownerService.js]: should be used ForbiddenError class

**Review by bartoszbialecki — APPROVED** [review]
Are you merged all current changes? Everything works?

**Pull request merged**